main.js

```diff
@@ -115,40 +128,22 @@ function uniquePalettes(palettesList, singlePalette) {
     return true;
 }
 
-function savePalette() {
-    if (!savedPalettes.length) {
-        savedPalettes.push(currentPalette)
-    } else if (uniquePalettes(savedPalettes, currentPalette)) {
-        savedPalettes.push(currentPalette)
-    }
-    displaySavedPalettesSection(savedPalettes);
-    getNewHexes(mainColorBoxes);
-}
-
 function deletePalette(savedPalette, savedPalettesIndex) {
     savedPalettes.splice(savedPalettesIndex, 1);
     savedPalette.remove();
+    if (!savedPalettes.length) {
+        paragraph.classList.remove('hidden');
+    }
 }
 
-function displaySavedPalettesSection(palette) {
-    savedPalettesSection.innerHTML = '';
-    if (!palette.length) {
-        paragraph.classList.remove('hidden');
-    } else {
-        paragraph.classList.add('hidden');
-        for (i = 0; i < palette.length; i++) {
-            savedPalettesSection.innerHTML += `
-            <div class="mini-container hover">
-                <div class="mini-box", style="background-color: #${palette[i][0]}"></div>
-                <div class="mini-box", style="background-color: #${palette[i][1]}"></div>
-                <div class="mini-box", style="background-color: #${palette[i][2]}"></div>
-                <div class="mini-box", style="background-color: #${palette[i][3]}"></div>
-                <div class="mini-box", style="background-color: #${palette[i][4]}"></div>
-                <img class="delete-button" src='./assets/delete.png'></img>
-            </div>
-            `
-        }
+function addPaletteToSavedPalettes(palette) {
+    var newMiniContainer = document.createElement('div');
+    newMiniContainer.classList.add('mini-container', 'hover');
+    savedPalettesSection.appendChild(newMiniContainer);
+    for (i = 0; i < palette.length; i++) {
+        newMiniContainer.innerHTML += `<div class="mini-box", style="background-color: #${palette[i]}"></div>`
```

Comment:
I like this! Good job! Add missing semicolon

Reply:
Agreed, this is a great refactor